### ISSUE TITLE
A 404 error was returned from the interface of nginx reverse proxy

### ISSUE THREAD

**Issue by lswzw** [rancher/rancher#29692]
**What kind of request is this (question):**

A 404 error was returned from the interface of nginx reverse proxy ,
rancher running by docker. 
GET /v3/connect/config HTTP/1.1 404

The above error is reported after adding a cluster. The interviews were normal, but it was strange to report an error.

**Steps to reproduce (least amount of steps as possible):**
docker-compost running
```
version: '3'
services:
  rancher:
    image: "rancher/rancher:v2.4.8"
    container_name: "rancher"
    restart: always
    entrypoint: entrypoint.sh --no-cacerts
    volumes:
      - ./rancherdata:/var/lib/rancher/
    networks:
      mynet:
        ipv4_address: 172.44.44.44
networks:
  mynet:
    external:
      name: rancher-network
#docker network create --driver=bridge --subnet=172.44.0.0/16 --ip-range=172.44.44.0/24 --gateway=172.44.44.1 rancher-network
```

NGINX 
```
map $http_upgrade $connection_upgrade {
    default Upgrade;
    ''      close;
}
server {
        listen     80;
        server_name  rancher.xxx.com;
        access_log  /data/nginx/logs/access_rancher.log  main;
        return 301 https://$server_name$request_uri;
}
server {
        listen 443 ssl;
        server_name  rancher.xxx.com;
        access_log  /data/nginx/logs/access_rancher.log  main;
        ssl_certificate /data/nginx/ssl/xxx.com/xxx.com.pem;
        ssl_certificate_key /data/nginx/ssl/xxx.com/xxx.com.key;
        error_page   500 502 503 504  /50x.html;
        location = /50x.html {
                root   html;
        }
        location / {
                proxy_set_header Host $host;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Forwarded-Port $server_port;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_pass http://172.44.44.44;
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection $connection_upgrade;
                # This allows the ability for the execute shell window to remain open for up to 15 minutes. Without this parameter, the default is 1 minute and will automatically close.
                proxy_read_timeout 300s;
                proxy_buffering off;
        }
}
```

**Result:**
```
xxxxxxx - - [22/Oct/2020:12:24:13 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
xxxxxxx - - [22/Oct/2020:12:24:15 +0800] "GET /v3/connect/config HTTP/1.1" 404 0 "-" "Go-http-client/1.1" "-"
```

**Comment by superseb** [comment]
please describe what `after adding a cluster` means, did you create a cluster (if so, what option) or did you import a cluster?

**Comment by lswzw** [comment]
Import an existing cluster Import an existing Kubernetes cluster.  Import local cluster。
```
[root@master01 ~]# kubectl get all -n cattle-system
NAME                                        READY   STATUS    RESTARTS   AGE
pod/cattle-cluster-agent-777fd8b4f5-hpxqm   1/1     Running   0          42h
pod/cattle-node-agent-7qlmc                 1/1     Running   0          42h
pod/cattle-node-agent-7rncv                 1/1     Running   0          42h

NAME                               DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/cattle-node-agent   7         7         7       7            7           <none>          42h

NAME                                   READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/cattle-cluster-agent   1/1     1            1           42h

NAME                                              DESIRED   CURRENT   READY   AGE
replicaset.apps/cattle-cluster-agent-777fd8b4f5   1         1         1       42h
replicaset.apps/cattle-cluster-agent-7cf94fdd98   0         0         0       42h
```

**Comment by superseb** [comment]
Merging into https://github.com/rancher/rancher/issues/15172, this should already be solved in 2.5
